R/trackhub.R

```diff
@@ -0,0 +1,69 @@
+### =========================================================================
+### TrackHub support
+### -------------------------------------------------------------------------
+
+### - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+### TrackHub class
+###
+
+setClass("TrackHub", representation(uri = "character"))
+
+isFileReference <- function(s) {
+  length(grep(".txt", s)) == 1
+}
+
+uri <- function(x, ...) x@uri
+
+getContent <- function(x) {
+  content <- readLines(x, warn = FALSE)
+  rexp <- "^(\\w+)\\s?(.*)$"
+  data.frame(field = sub(rexp, "\\1", content),
+             value = sub(rexp,"\\2", content))
+}
+
+getValue <- function(field, content) {
+  position <- grep(field, content$field)
+  content$value[position]
+}
+
+hubFile <- function(x) file.path(uri(x), "hub.txt")
+
+genomesFile <- function(x, y) file.path(uri(x), y)
+
+setMethod("genome", "TrackHub", function(x) {
+  hubContent <- getContent(hubFile(x))
+  field <- "genomesFile"
+  genomesFileValue <- getValue(field, hubContent)
+  if (isFileReference(genomesFileValue)) {
+    path <- genomesFile(x, genomesFileValue)
+    genomeContent <- getContent(path)
+    genomes <- getValue("genome", genomeContent)
+  }
+  else genomes <- genomesFileValue
+  as.character(structure(genomes, name = field))
```

Comment:
Why attach a `name` attribute here?

Reply:
I think I was using it for debugging this somehow made into the commit